token/cli/src/clap_app.rs

```diff
@@ -802,7 +839,7 @@ pub fn app<'a, 'b>(
                         .number_of_values(1)
                         .conflicts_with("transfer_fee")
                         .requires("transfer_fee_basis_points")
-                        .validator(is_amount)
+                        .value_parser(clap::value_parser!(f64))
```

Comment:
can this parse a number without a decimal?

Reply:
Yes you are completely right! The validation `is_amount` should actually be replaced with using the `Amount` [type](https://github.com/anza-xyz/agave/blob/master/clap-v3-utils/src/input_parsers/mod.rs#L142), but I thought making this change in this PR would add on a whole new set of changes, so I just tried using the built-in clap value parser without thinking through it carefully. Thanks for the catch!